packages/tool_util/HISTORY.rst

```diff
@@ -5,6 +5,17 @@ History
 
 .. to_doc
 
+22.1.2 (2022-08-29)
+-------------------
+
+* Fix lint context error level
+* Pin galaxy-util to >= 22.1
+
+22.1.1 (2022-08-22)
+--------------------
+
+* First release from the 22.01 branch of Galaxy
```

Comment:
Thanks. Any hint on how to where to look for the API breaking changes in 22.01?

https://github.com/common-workflow-language/cwltool/issues/1722

Reply:
https://github.com/galaxyproject/galaxy/commit/da61e77068f115ebe0afcd10a6d34b9dd43aaae7 ? That looks broken, no idea what I wanted to do with that `except KeyError`

Reply:
ah.. So cwltool should just pin an earlier version of galaxy-tool-util until this gets fixed?

We also have the problem that the galaxy biocontainer resolver is not pulling from biocontainers, but is instead mulling an entire new container locally. This is very slow!

Reply:
That would certainly explain why you ran into this error in the first place. Let me have a look at cwltool

Reply:
OK, I took the core of https://github.com/common-workflow-language/cwltool/blob/9eeccee69df2b3f0a70bc414616446bdfcc6c952/cwltool/software_requirements.py#L133-L156 and turned that into a unit test, and now I also understand why I worked with `KeyError`, it's not as crazy as it seems (it's a beaker object) but we should have definitely had type annotations.

Reply:
Cool. Any hints as to why we are getting locally mulled containers instead of a quay.io biocontainer reference?

Reply:
Not yet, but the unit test is doing that too, so I'm going to fix that